perl-package/AI-MXNet/lib/AI/MXNet/NDArray.pm

```diff
@@ -384,8 +385,11 @@ method _slice (
 )
 {
     confess("start $start > stop $stop") if $start > $stop;
+    my $sub = AI::MXNet::RunTime->Features()->is_enabled('INT64_TENSOR_SIZE')
+              ? \&AI::MXNetCAPI::NDArraySlice64
+              : \&AI::MXNetCAPI::NDArraySlice;
```

Comment:
It seems a little awkward to put the caller in charge of checking the `INT64_TENSOR_SIZE` compile setting.  Would it make sense to make `AI::MXNetCAPI::NDArraySlice()` and friends responsible for checking the flag and deferring to the `NDArray*64()` variants when appropriate?

Reply:
@tlby On python side the switch is made in the caller space. The c api has two functions defined for each task, one with ```(u)?int32_t``` input/output and another with `(u)?int64_t`
I don't know how to combine these two functions via the swig space.
Looks like a task for you to complete later the line.

Reply:
If that's how things are done in Python, then no need to diverge from this pattern.  Thanks!